src/goto-instrument/code_contracts.cpp

```diff
@@ -188,6 +205,37 @@ void code_contractst::check_apply_invariants(
     converter.goto_convert(assumption, havoc_code, mode);
   }
 
+  // Temproary variables that store the variant before and after the loop
+  symbol_exprt old_variant_symbol =
+    new_tmp_symbol(
+      variant.type(), loop_head->source_location, function_name, mode)
+      .symbol_expr();
+  symbol_exprt new_variant_symbol =
+    new_tmp_symbol(
+      variant.type(), loop_head->source_location, function_name, mode)
+      .symbol_expr();
+
+  if(!variant.is_nil())
+  {
+    // Generate: a declaration of the temporary variable that stores the loop
+    // variant before the loop entry
+    havoc_code.add(
+      goto_programt::make_decl(old_variant_symbol, loop_head->source_location));
+
+    // Generate: an assignment to store the loop variant before the loop entry
+    // We would like to set a comment associated with a source location.
+    // However, we have already used loop_head->source_location to set a
+    // comment. Furthermore, the third argument of make_assignment is passed by
+    // reference rather than by value. Therefore, it is necessary to clone
+    // loop_head->source_location.
```

Comment:
I don't understand the issue here. What exactly is the problem with passing the `source_location` by value? Is it that we are adding a comment? If so, please remove the comment on the assignment (I explained the rationale in another comment below).

(same comment for the other assignment below)

Reply:
Now that we use `goto_convert` instead of `make_assignment`, we no longer need to clone a `source_location` object. So this issue is now gone.